File: 71. Simplify Path/71. Simplify Path.go
func simplifyPath(path string) string {
    stack := []string{}
    parts := strings.Split(path, "/")
    
    for _, part := range parts {
        if part == "." || part == "" {
            continue
        }
        if part == ".." {
            if len(stack) > 0 {
                stack = stack[:len(stack)-1]
            }
        } else {
            stack = append(stack, part)
        }
    }
    
    return "/" + strings.Join(stack, "/")
}

File: 71. Simplify Path/71. Simplify Path.dart
class Solution {
  String simplifyPath(String path) {
    final stack = <String>[];
    final parts = path.split('/');
    
    for (final part in parts) {
      if (part == '.' || part.isEmpty) continue;
      if (part == '..') {
        if (stack.isNotEmpty) stack.removeLast();
      } else {
        stack.add(part);
      }
    }
    
    return '/' + stack.join('/');
  }
}